class DefaultLegacySupportTest$MyRunnable extends java.lang.Object implements java.lang.Runnable
{
    private volatile org.apache.maven.execution.MavenSession session;
    final org.apache.maven.plugin.internal.DefaultLegacySupportTest this$0;

    void <init>(org.apache.maven.plugin.internal.DefaultLegacySupportTest)
    {
        DefaultLegacySupportTest$MyRunnable this;
        org.apache.maven.plugin.internal.DefaultLegacySupportTest this$0;
        org.apache.maven.plugin.internal.DefaultLegacySupportTest$MyRunnable $r0;

        this := @this: DefaultLegacySupportTest$MyRunnable;

        this$0 := @parameter0: org.apache.maven.plugin.internal.DefaultLegacySupportTest;

        $r0 = (org.apache.maven.plugin.internal.DefaultLegacySupportTest$MyRunnable) this;

        $r0.<org.apache.maven.plugin.internal.DefaultLegacySupportTest$MyRunnable: org.apache.maven.plugin.internal.DefaultLegacySupportTest this$0> = this$0;

        specialinvoke this.<java.lang.Object: void <init>()>();

        return;
    }

    public void run()
    {
        DefaultLegacySupportTest$MyRunnable this;
        org.apache.maven.plugin.internal.DefaultLegacySupportTest $stack2, $stack4;
        java.util.concurrent.CountDownLatch $stack3;
        org.apache.maven.plugin.internal.DefaultLegacySupport $stack5;
        org.apache.maven.execution.MavenSession $stack6;
        java.lang.InterruptedException $stack7;
        org.apache.maven.plugin.internal.DefaultLegacySupportTest$MyRunnable $r0, $r1, $r2;

        this := @this: DefaultLegacySupportTest$MyRunnable;

     label1:
        $r0 = (org.apache.maven.plugin.internal.DefaultLegacySupportTest$MyRunnable) this;

        $stack2 = $r0.<org.apache.maven.plugin.internal.DefaultLegacySupportTest$MyRunnable: org.apache.maven.plugin.internal.DefaultLegacySupportTest this$0>;

        $stack3 = $stack2.<org.apache.maven.plugin.internal.DefaultLegacySupportTest: java.util.concurrent.CountDownLatch latch>;

        virtualinvoke $stack3.<java.util.concurrent.CountDownLatch: void await()>();

     label2:
        goto label4;

     label3:
        $stack7 := @caughtexception;

     label4:
        $r1 = (org.apache.maven.plugin.internal.DefaultLegacySupportTest$MyRunnable) this;

        $stack4 = $r1.<org.apache.maven.plugin.internal.DefaultLegacySupportTest$MyRunnable: org.apache.maven.plugin.internal.DefaultLegacySupportTest this$0>;

        $stack5 = $stack4.<org.apache.maven.plugin.internal.DefaultLegacySupportTest: org.apache.maven.plugin.internal.DefaultLegacySupport defaultLegacySupport>;

        $stack6 = virtualinvoke $stack5.<org.apache.maven.plugin.internal.DefaultLegacySupport: org.apache.maven.execution.MavenSession getSession()>();

        $r2 = (org.apache.maven.plugin.internal.DefaultLegacySupportTest$MyRunnable) this;

        $r2.<org.apache.maven.plugin.internal.DefaultLegacySupportTest$MyRunnable: org.apache.maven.execution.MavenSession session> = $stack6;

        return;

        catch java.lang.InterruptedException from label1 to label2 with label3;
    }

    public org.apache.maven.execution.MavenSession getSession()
    {
        DefaultLegacySupportTest$MyRunnable this;
        org.apache.maven.execution.MavenSession $stack1;
        org.apache.maven.plugin.internal.DefaultLegacySupportTest$MyRunnable $r0;

        this := @this: DefaultLegacySupportTest$MyRunnable;

        $r0 = (org.apache.maven.plugin.internal.DefaultLegacySupportTest$MyRunnable) this;

        $stack1 = $r0.<org.apache.maven.plugin.internal.DefaultLegacySupportTest$MyRunnable: org.apache.maven.execution.MavenSession session>;

        return $stack1;
    }
}
